src/types/mock.rs

```diff
@@ -26,11 +29,11 @@ impl Arbitrary for MockNetwork {
         let edges = arbitrary_normalised_edges_from(g, &nodes);
 
         for n in &nodes {
-            graph.add_node(n.id().clone(), n.get_metadata().clone())
+            graph.add_node(n.id().clone(), n.data().clone())
         }
 
         for e in edges {
-            graph.add_edge(e.source, e.target, e.id, e.metadata)
+            graph.add_edge(e.id, e.source, e.target, e.weight, e.metadata)
```

Comment:
Could we rename `metadata` for edges to `data` to be consistent with `data` for nodes?

Reply:
Yup, good point, let me.